lib/scoring.rb

```diff
@@ -0,0 +1,40 @@
+module Scrabble
+  class Scoring
+    LETTER_SCORES = {"a" => 1, "c" => 3, "b" => 3, "e" => 1, "d" => 2, "g" => 2,
+             "f" => 4, "i" => 1, "h" => 4, "k" => 5, "j" => 8, "m" => 3,
+             "l" => 1, "o" => 1, "n" => 1, "q" => 10, "p" => 3, "s" => 1,
+             "r" => 1, "u" => 1, "t" => 1, "w" => 4, "v" => 4, "y" => 4,
+             "x" => 8, "z" => 10}
+
+    def self.tie_breaker(tie_words)
+      tie_words.each do |word|
+        if word.length == 7
+          return word
+        else
+          # The & calls to_proc on the object
+          # Returns a proc object that expects a parameter and calls a method
+          # parameter is the tie_words and the method is length
+          return tie_words.min_by(&:length)
```

Comment:
Good work doing the research on procs. However, I'm not sure this implementation will work - I think line 17 needs to be outside the `each` loop. As a test case, try `['zzzzzz', 'aaaaaah']`